spark/sql-20/src/main/scala/org/elasticsearch/spark/sql/Utils.java

```diff
@@ -44,6 +44,9 @@ static FieldType extractType(Field field) {
     // double filtering (run Spark filters) or not
     static final String DATA_SOURCE_KEEP_HANDLED_FILTERS = "es.internal.spark.sql.pushdown.keep.handled.filters";
 
+    // columns selected by Spark SQL query
+    static final String DATA_SOURCE_REQUIRED_COLUMNS = "es.internal.spark.sql.required.columns";
```

Comment:
Could we change this to a different location? Like `InternalConfigurationOptions`?